lib/token.js

```diff
@@ -136,14 +170,13 @@ function handleOAuthResponse(sdk, oauthParams, res, urls) {
 
   return new Q()
   .then(function() {
-    if (res['error'] || res['error_description']) {
-      throw new OAuthError(res['error'], res['error_description']);
-    }
-
-    if (res.state !== oauthParams.state) {
-      throw new AuthSdkError('OAuth flow response state doesn\'t match request state');
+    validateResponse(res, oauthParams);
+    if (res['code']) {
+      tokenTypes = pkce.loadMeta(sdk).responseType;
+      return exchangeCodeForToken(sdk, oauthParams, res['code'], urls);
```

Comment:
Also noted earlier, we need to account for responses that contain the following combinations supported by PKCE:
- `code`
- `code token`
- `code id_token`

Reply:
As we discussed, "hybrid" scenarios like this will not be supported at this time.